quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -1391,19 +1391,28 @@ export class CompilerVisitor implements IRVisitor {
     this.compStack.push(mkFunComputable(lazyCompute))
   }
 
-  // Apply the operator oneOf.
+  // Apply the operator oneOf
   private applyOneOf(sourceId: bigint) {
     this.applyFun(sourceId, 1, set => {
-      const sizeOrNone = set.cardinality()
-      if (sizeOrNone.isJust()) {
-        if (sizeOrNone.value === 0n) {
-          this.errorTracker.addRuntimeError(sourceId, `Applied oneOf to an empty set`)
-          return none()
-        }
-        return set.pick(this.rand(sizeOrNone.value))
-      }
-      // an infinite set, pick an integer from the range [-2^255, 2^255)
-      return set.pick(-(2n ** 255n) + this.rand(2n ** 256n))
+      const bounds = set.bounds()
+      const positions: bigint[] = bounds.map(b => {
+        return (
+          b
+            .map(sz => {
+              if (sz === 0n) {
+                this.errorTracker.addRuntimeError(sourceId, `Applied oneOf to an empty set`)
+              }
+              return this.rand(sz)
+            })
+            // An infinite set, pick an integer from the range [-2^255, 2^255).
+            // TODO: make it a configurable parameter:
+            // https://github.com/informalsystems/quint/issues/279
+            .or(just(-(2n ** 255n) + this.rand(2n ** 256n)))
```

Comment:
Can't the codomain be `Nat` and shouldn't the lower bound then be non-negative?

Reply:
Yep, the implementation of `pick` for Int/Nat checks that and uses the absolute value in case of `Nat`. I just do not want to differentiate them here.

Reply:
Okay. Maybe leave a comment then, so it's not confusing in the future?